server/schema/blog.js

```diff
@@ -20,21 +20,25 @@ const BlogSchema = z.object({
     datePublished: z.string().datetime(),
 })
```

Comment:
We have to get rid of id, datePublished and authorId from the schema because these aren't things we receive from the user.

- The authorId, we will get by fetching the current logged-in user. The ID of the current user will be the authorID.
- The datePublished will be generated by the database when the post is added to the database.
- The id is also generated by the database when the post is created.

Reply:
good points, `id` however is blog ID which we thought we would generate but now I think slug would do the job. Regardless, here is the new schema:
```javascript
const BlogSchema = z.object({
    title: z.string().min(1).max(120),
    category: z.enum([
        "AI",
        "BLOCKCHAIN",
        "CLOUD",
        "DEVOPS",
        "METAVERSE",
        "NFT",
        "WEB3",
    ]),
    description: z.string().max(256),
    content: z.string().min(1).max(16384),
    imageUrl: z.string().url(),
})
```

Reply:
The ID is always generated by the database. The slug yes, you would have to generate that yourself. But dw about adding the slug to the validation schema. This schema looks good so far!

To improve it further, you can also set a minimum value for the description and maybe increase the content length? For example, on my blog the minimum content length is 512 characters. Also custom error messages if you wanna do better error handling.

For example:

```
title: z.string().min(3, {message: "Title must be at least 3 characters in length."})
```

Without the above custom message, the error would just say: String must be 3 characters in length.

That isn't a bad error message but we gotta remember that not everyone knows what a string is, especially if they're not familiar with software terms.

Reply:
oh Kodi is working in having a structured error handling. @kodishim could you implement this in the error handling classes that you have created? or should I do it in zod itself?

Reply:
It's better to do it in zod itself and then get the Zod error message from the custom error class he's building.